syntax = "proto3";

package brightness;

// handles sky brightness prediction and pollution requests
service BrightnessService {
  rpc GetBrightnessObservation (Coordinates) returns (BrightnessObservation) {}
  rpc GetPollution (Coordinates) returns (Pollution) {}
}

message Coordinates {
  double lat = 1;
  double lon = 2;
}

message Pollution {
  uint32 r = 1;
  uint32 g = 2;
  uint32 b = 3;
  uint32 a = 4;
}

message BrightnessObservation {
  string uuid = 1;
  double lat = 2;
  double lon = 3;
  string utc_iso = 4;
  float mpsas = 5;
}
